cmake/atomic_configure_file.cmake

```diff
@@ -1,9 +1,11 @@
 function(atomic_configure_file input output)
-  set(atomic_file "${CMAKE_BINARY_DIR}/atomic_configure_file")
+  get_filename_component(atomic_file ${output} NAME)
+  set(atomic_file "${CMAKE_BINARY_DIR}/${atomic_file}")
```

Comment:
To reduce the chance of collisions if the basename of `output` is the same as an existing file in the binary dir please keep `atomic_configure_file` as a prefix to the name: `set(atomic_file "${CMAKE_BINARY_DIR}/atomic_configure_file_${atomic_file}")`

Reply:
This was changed intentionally. Using a different name than the final one, the copying needs two steps: first copying and then renaming the file at the destination location. (Remember that COPY doesn't allow for a simultaneous rename).
For me, this contradicts the atomic idea.

Reply:
Good point - I didn't consider that. Can you place the file in a subdirectory `atomic_configure_file` then to minimize the chance of collisions.